plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerRegistryChecker.java

```diff
@@ -18,42 +24,56 @@
 import javax.inject.Named;
 import javax.inject.Singleton;
 import java.io.IOException;
-import java.net.HttpURLConnection;
-import java.net.URL;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static java.lang.Boolean.TRUE;
 
 /**
  * Checks that docker registry is available.
  *
  * @author Yevhenii Voevodin
+ * @author Alexander Andrienko
  */
 @Singleton
 public class DockerRegistryChecker {
 
     private static final Logger LOG = LoggerFactory.getLogger(DockerRegistryChecker.class);
 
+    private final HttpJsonRequestFactory requestFactory;
+    private final String                 registryUrl;
+    private final Boolean                snapshotUseRegistry;
+
     @Inject
-    @Named("docker.registry.auth.url")
-    private String registryUrl;
+    public DockerRegistryChecker(HttpJsonRequestFactory requestFactory,
+                                 @Nullable @Named("machine.docker.registry") String registryUrl,
+                                 @Nullable @Named("machine.docker.snapshot_use_registry") Boolean snapshotUseRegistry) {
+        this.requestFactory = requestFactory;
+        this.registryUrl = registryUrl;
+        this.snapshotUseRegistry = snapshotUseRegistry;
+    }
 
     /**
      * Checks that registry is available and if it is not - logs warning message.
      */
+    @VisibleForTesting
     @PostConstruct
-    private void checkRegistryIsAvailable() throws IOException {
-        LOG.info("Probing registry '{}'", registryUrl);
-        final HttpURLConnection conn = (HttpURLConnection) new URL(registryUrl).openConnection();
-        conn.setConnectTimeout(30 * 1000);
-        try {
-            final int responseCode = conn.getResponseCode();
-            LOG.info("Probe of registry '{}' succeed with HTTP response code '{}'", registryUrl, responseCode);
-        } catch (IOException ioEx) {
-            LOG.warn("Docker registry " + registryUrl + " is not available, " +
-                     "which means that you won't be able to save snapshots of your workspaces." +
-                     "\nHow to configure registry?" +
-                     "\n\tLocal registry  -> https://docs.docker.com/registry/" +
-                     "\n\tRemote registry -> set up 'docker.registry.auth.*' properties");
-        } finally {
-            conn.disconnect();
+    void checkRegistryIsAvailable() {
+        if (TRUE.equals(snapshotUseRegistry) && !isNullOrEmpty(registryUrl)) {
+            String registryLink = "http://" + registryUrl;
```

Comment:
it's strange to have http:// being concatenated with a variable named "URL" because URL is expected to already provide the protocol